app/services/stats/calculations.rb

```diff
@@ -78,5 +78,58 @@ def self.calculate_percentage(value, total)
       end
     end
 
+      # this method is passed into the stats collector as a before_finalize callback and accesses the
+      # @stats variable  inside the stats collector to sum totals for directorates and business groups
+      def self.roll_up_stats_callback(stats)
+        overall_total_results = stats.stats[:total]
+        BusinessUnit.includes(:directorate, :business_group).all.each do |bu|
+          bu_results = stats.stats[bu.id]
+          directorate_results = stats.stats[bu.directorate.id]
+          business_group_results = stats.stats[bu.business_group.id]
+          bu_results.each do |key, value|
+            directorate_results[key] += value
+            business_group_results[key] += value
+            overall_total_results[key] += value
+          end
+        end
+      end
+
+      def self.populate_team_details_callback(stats)
+        # This is a hash (indexed by team id) containing a hash of team stats columns
```

Comment:
I think this could also possibly be split up - the branches of the case statement could be methods or the case statement could be extracted 🤔 

I'm also think the 'totals' section at the bottom could be its own method 👍

Reply:
@willmcb  as I said, this part was not written by me and I didn't change it, so I would suggest to leave it as it is